packages/sfu/server.ts

```diff
@@ -0,0 +1,121 @@
+}
+
+async function runMediasoupWorkers() {
+  const {numWorkers} = config.mediasoup
+  const {logLevel, logTags, rtcMinPort, rtcMaxPort} = config.mediasoup.workerSettings
+  for (let i = 0; i < 1; i++) {
+    // hard code to 1 for now
+    const worker = await createWorker({
+      logLevel: logLevel as mediasoupTypes.WorkerLogLevel,
+      logTags: logTags as mediasoupTypes.WorkerLogTag[],
+      rtcMinPort: Number(rtcMinPort),
+      rtcMaxPort: Number(rtcMaxPort)
+    })
+    worker.on('died', () => {
+      logger.info('mediasoup Worker died, exiting  in 2 seconds... [pid:%d]', worker.pid)
+      setTimeout(() => process.exit(1), 2000)
+    })
+    mediasoupWorkers.push(worker)
+    // setInterval(async () => {
+    //   const usage = await worker.getResourceUsage()
+    //   logger.info('mediasoup Worker resource usage [pid:%d]: %o', worker.pid, usage)
+    // }, 120000)
+  }
+}
+
+async function runWebSocketServer() {
+  const isProduction = process.env.NODE_ENV === 'production'
+  const httpModule = isProduction ? await import('http') : await import('https')
+  const options = isProduction
+    ? {}
+    : {
+        cert: fs.readFileSync(config.https.tls.cert),
+        key: fs.readFileSync(config.https.tls.key)
+      }
+  const server = httpModule.createServer(options, express() as any)
+
+  await new Promise((resolve) => {
+    server.listen(Number(config.http.listenPort), config.http.listenIp, resolve)
+  })
+  const websocketServer = new protoo.WebSocketServer(server, {
+    maxReceivedFrameSize: 960000, // 960 KBytes.
+    maxReceivedMessageSize: 960000,
+    fragmentOutgoingMessages: true,
+    fragmentationThreshold: 960000
+  })
+  console.log(`\n🎥🎥🎥 Media Server Ready: Port ${config.http.listenPort}  🎥🎥🎥`)
+  // validate auth token
+  websocketServer.on('connectionrequest', async (info, accept, reject) => {
+    const requestUrl = url.parse(info.request.url, true)
+    const {roomId, peerId, authToken: encodedAuthToken, teamId} = requestUrl.query
+    const missingParams = [] as string[]
+    if (!roomId) missingParams.push('roomId')
+    if (!peerId) missingParams.push('peerId')
+    if (!teamId) missingParams.push('teamId')
+    if (!encodedAuthToken) missingParams.push('authToken')
+    if (missingParams.length > 0) {
+      reject(400, 'Connection request without required field(s):', missingParams)
+      return
+    }
+    const decodedAuthToken = getVerifiedAuthToken(encodedAuthToken as string)
+    if (!isAuthenticated(decodedAuthToken)) {
+      reject(401, 'No authentication credentials')
+      return
+    }
+    const {sub: userId, iat} = decodedAuthToken
+    const isBlacklistedJWT = await checkBlacklistJWT(userId, iat)
+    logger.debug('Is blacklisted:', isBlacklistedJWT)
+    if (isBlacklistedJWT) {
+      reject(401, 'Your authentication has expired')
+      return
+    }
+    if (!isTeamMember(decodedAuthToken, teamId as string)) {
+      reject(403, "Oops! You're not authorized to be here")
```

Comment:
we auth the teamId, but we don't auth the roomId.
if an attacker has the roomId could they join the chat?
we could prefix the `roomId` with the `teamId`, e.g. `roomId = `${teamId}:foo123`
alternatively, we could let the server generate the roomId itself.
i'm sure other folks have solved this & the nuances (e.g. having different clients race to create or get the room). it'd be fun to check out what those solutions are.

Reply:
this is very true. it's currently only doing authentication but not authorization for the room. i like the idea of prefixing the roomId with the teamId.

Reply:
the `roomId` is now is composite of of `teamId` and `meetingId`